extensions/music/src/play-library-album.tsx

```diff
@@ -1,85 +1,101 @@
-import { Action, ActionPanel, closeMainWindow, List, showToast, Toast, ToastStyle, useNavigation } from "@raycast/api";
+import { Grid, List, Action, ActionPanel, closeMainWindow, showToast, Toast, useNavigation } from "@raycast/api";
+import { ListOrGrid, getViewLayout, getGridItemSize, getMaxNumberOfResults } from "./util/listorgrid";
+import { getArtworkByIds } from "./util/scripts/track";
 import { flow, pipe } from "fp-ts/lib/function";
 import * as O from "fp-ts/Option";
 import * as S from "fp-ts/string";
 import * as T from "fp-ts/Task";
 import * as TE from "fp-ts/TaskEither";
-import React, { useEffect, useState } from "react";
+import { useEffect, useState } from "react";
 import { Album } from "./util/models";
 import { fromEmptyOrNullable } from "./util/option";
 import { parseResult } from "./util/parser";
 import * as music from "./util/scripts";
 
 export default function PlayLibraryAlbum() {
-  const [albums, setAlbums] = useState<readonly Album[] | null>(null);
+  const [albums, setAlbums] = useState<readonly Album[] | null>([]);
+  const [artworks, setArtworks] = useState<any>({});
   const { pop } = useNavigation();
-
-  const loadAll = pipe(
-    music.albums.getAll,
-    TE.map(parseResult<Album>()),
-    TE.matchW(
-      () => {
-        showToast(Toast.Style.Failure, "Could not get albums");
-        return [] as ReadonlyArray<Album>;
-      },
-      flow(
-        fromEmptyOrNullable,
-        O.matchW(() => setAlbums([]), setAlbums)
-      )
-    )
-  );
-
-  useEffect(() => {
-    loadAll();
-  }, []);
+  const layout = getViewLayout();
+  const numResults = getMaxNumberOfResults();
+  const gridItemSize = getGridItemSize();
 
   const onSearch = async (next: string) => {
-    setAlbums(null); // start loading
-
-    if (!next || next?.length < 1) {
-      setAlbums(null);
-      await loadAll();
-      return;
-    }
+    if (!next) return;
+    // start loading
+    setAlbums(null);
+    setArtworks(null);
 
     await pipe(
       next,
       S.trim,
-      music.track.search,
+      music.albums.search,
       TE.matchW(
         () => {
           showToast(Toast.Style.Failure, "Could not get albums");
           return [] as ReadonlyArray<Album>;
         },
-        (tracks) =>
-          pipe(
-            tracks,
+        (albums) => {
+          albums = albums.replace(/\s&\s/g, " and ");
+          albums = albums.split("\n").slice(0, numResults).join("\n");
+          return pipe(
+            albums,
             fromEmptyOrNullable,
             O.matchW(() => [] as ReadonlyArray<Album>, parseResult<Album>())
-          )
+          );
+        }
       ),
       T.map(setAlbums)
     )();
   };
 
+  useEffect(() => {
```

Comment:
as in the rest of the extension I would replace `Promises` with `TaskEither`. But I understand that the library might be a wall for new contributors.

Reply:
Yeah I didn't really understand how to use `TaskEither`. Feel free to make the conversion if it is simple.